### PR TITLE
Feature/4101 update ck editor version

### LINKED ISSUE #4101: Update CKEditor version
Referencing https://github.com/DNN-Connect/CKEditorProvider/issues/87

This issue is about updating CKEditor version. Moving it here since we are merging https://github.com/DNN-Connect/CKEditorProvider in the main repository here.

I leave it to be triaged until we can discuss this more or someone picks up this task :)

### PR DESCRIPTION
**Pull request by skamphuis**
Closes #4101 

## Summary
I updated CKEditor to version 4.15.1 and made the required/appropriate changes for that:
- moved DNN icons and images out of CKEditor folder
- changed "broken" javascript in EditHTML.ascx for resizing the editor after window.resize
- updated version numbers for CKEditor in code (mainly for file paths)

### CODE REVIEW

**Review by bdukes — APPROVED** [review]
Thanks!

This will certainly need some real world testing, but all of the changes look good.  

Once #4374 is merged, I have a branch where I've rebased this PR on those changes. Thanks for the very clean commits, it made the rebase very straightforward :-)

**Review by donker — APPROVED** [review]
Thanks for working on this editor provider!

**Comment by bdukes** [comment]
FYI, I've pushed the rebased commits to your branch since #4374 was merged.

**Comment by skamphuis** [comment]
While testing some more today, I noticed 2 plugins and a skin where missing from the updated CKE:
* plugin dnnpages
* plugin wordcount
* skin moono (used for the popups)

**Comment by skamphuis** [comment]
Another two commits needed after testing. Link popup didn't seem to work.

- Had to remove the easyimage and cloudservice plugins, to prevent console warnings (those plugins are not included, but seem to be known in CKE anyway)
- DefaultLinkProtocol property (new in CKE 4.11) is needed for the link popup to work.

File: DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Utilities/Utility.cs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information
 
+using DNNConnect.CKEditorProvider.Constants;
```
Comment (bdukes) [inline DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Utilities/Utility.cs:5]: (nit) Can this get moved inside the `namespace` declaration, please?

Comment (skamphuis) [reply DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Utilities/Utility.cs:5]: Sure, I'll fix that. 

I noticed changes like this recently. Do you know the benefits of that?

Comment (bdukes) [reply DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Utilities/Utility.cs:5]: It's primarily a style rule, but it does have some minor benefits ([described in the StyleCop rule docs](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/fe00f94f0ba5471eb9e3e752a3ae37bf19e5f2e3/documentation/SA1200.md))

Comment (skamphuis) [reply DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Utilities/Utility.cs:5]: That makes sense, thanks.

**Comment by bdukes** [comment]
Thanks!

**Pull request merged**